### PR TITLE
Enable native test binaries to be built for Arm64

### PR DESCRIPTION
**Pull request by ramarag**
@gkhanna79 @rahku  PTAL

### CODE REVIEW

**Comment by ramarag** [comment]
@dotnet-bot test windows_nt arm64

**Comment by ramarag** [comment]
@dotnet-bot test CI please

File: netci.groovy [hunk 1]
```diff
@@ -844,14 +844,14 @@ combinedScenarios.each { scenario ->
                                     break
                                 case 'arm64':
                                     assert scenario == 'default'
-                                    buildCommands += "set __TestIntermediateDir=int&&build.cmd ${lowerConfiguration} ${architecture} skiptestbuild /toolset_dir C:\\ats"
+                                    buildCommands += "set __TestIntermediateDir=int&&build.cmd ${lowerConfiguration} ${architecture} /toolset_dir C:\\ats"
```
Comment (gkhanna79) [inline netci.groovy]: I would suggest that you submit this change seperately to test the CI once the primary change has been committed.

**Comment by gkhanna79** [comment]
Overall, looks good modulo comments. PTAL at the x86 build failure and also request a Arm64 CI build to ensure that is clean prior to merge.

**Comment by ramarag** [comment]
@dotnet-bot  test Windows_NT x86 Release Legacy Backend Build and Test

**Comment by ramarag** [comment]
The Arm64 CI builds are listed below are there any other jobs that are required to be triggered ?

File: netci.groovy (hunk 1 above)
Comment (ramarag) [inline netci.groovy]: The CI changes in the netci.groovy  are not reflected in the PR carying the change. So just checking in the change without above will not be of any use till the above goes in

**Comment by ramarag** [comment]
@dotnet-bot test windows_nt arm64

**Comment by rahku** [comment]
LGTM. I would just ensure that one of the native tests actually run fine.

**Comment by ramarag** [comment]
Already tested out a candidate interop test

**Comment by ramarag** [comment]
@dotnet-bot test CI

**Comment by rahku** [comment]
On thinking more about this. This does not look right. buildtest.cmd seems to be taking a dependency on build.cmd This would not allow one to run buildtests.cmd directly for arm64....which you need to do when building pri1 tests.

**Comment by jashook** [comment]
Seems like this change is causing the ci system to fail build Arm64 cross builds. http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/arm64_cross_release_windows_nt_rc2/31/console

**Comment by ramarag** [comment]
I am not making any change here build.cmd skiptests skipnative should build the tests as is

**Comment by ramarag** [comment]
I am not making any change here build.cmd skipmscorlib skipnative should build the tests as is

**Comment by gkhanna79** [comment]
@ramarag This PR broke the build. If you look at the Release Arm64 build done by the PR (http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/arm64_cross_release_windows_nt_prtest/18/console), it skipped tests since netci.groovy change didnt come into affect and used the last setting in CI,which was to skip the test build.

Post PR, which included your netci.groovy change, CI is now building the tests as part of the build. (http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/arm64_cross_release_windows_nt_rc2/31/console). Clearly, something is wrong since test build of CI is broken. This was one of the reason you were requested to decouple the CI change from PR and test it independently as the regression would have come to light.

Please look into fixing this at the earliest.

**Comment by ramarag** [comment]
This PR did not break the build,  There was a subsequent PR https://github.com/dotnet/coreclr/issues/3578, which was tested clean before this PR hit mainline which is the actual reason for the break [ last one to make a change that broke build]

the fix is being taken care by https://github.com/dotnet/coreclr/pull/3640

**Comment by gkhanna79** [comment]
The fix you point does not explain the following message in CI:

> C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.Cpp.InvalidPlatform.Targets(22,7): error MSB8007: The Platform for project 'install.vcxproj' is invalid.  Platform='arm64'

**Comment by ramarag** [comment]
This change is Rc2, unfortunately it also uses the netci.groovy of the master which is causing the build issue. https://github.com/dotnet/coreclr/pull/3197 appears to be solving this issues by picking up branch specific netci.groovy

@mmitche  is planning to merge some time late this week, but he is willing to do it earlier if required

**Comment by ramarag** [comment]
@mmitche  has deiced to prioritize the fix for CI, so i am not doing a stop gap fix.
The only option for the stop gap fix as of now is to roll back the CI changes i did

**Pull request merged**